<html>
<div id="root"></div>
<script src="https://unpkg.com/react@16.12.0/umd/react.development.js"></script>
<script src="https://unpkg.com/react-dom@16.12.0/umd/react-dom.development.js"></script>
<script src="https://unpkg.com/@babel/standalone@7.8.3/babel.min.js"></script>
<script src="https://unpkg.com/prop-types@15.6.1/prop-types.min.js"></script>
<script src="https://unpkg.com/react-error-boundary@1.2.3/dist/umd/react-error-boundary.js"></script>
<script type="text/babel">
    const rootElement = document.getElementById('root')

    const ErrorBoundary = ReactErrorBoundary.ErrorBoundary

    const ErrorFallBack = ({error}) =>
        <div>
            <p>Something went </p>
            <p>{error.message}</p>
        </div>

    const Bomb = () => {
        throw new Error("*--O Boom !")
    }

    const App = () => {
        const [explode, setExplode] = React.useState(false)
        return (
            <div>
                <button onClick={() => setExplode(true)}>Explode</button>
                <br/>
                <ErrorBoundary FallbackComponent={ErrorFallBack}>
                    {explode ? <Bomb/> : 'Push the button'}
                </ErrorBoundary>
            </div>
        )
    }

    ReactDOM.render(<App/>, rootElement)
</script>
</html>